sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_ptr.asciidoc

```diff
@@ -613,6 +706,46 @@ Increment and decrement operator of annotated_ref. Increment/Decrement the objec
 referenced by this wrapper via ``T``'s Increment/Decrement operator.
 
 The annotations are applied when the object `T` is loaded and stored to the memory.
+
+a|
+[source,c++]
+----
+template <typename T2, typename PropertyList2,
+          typename R = decltype(std::declval<T>() OP std::declval<T2>())>
+R operatorOP(const annotated_ref<T2, PropertyList2> &other) const;
+----
+|
+Defines the overloaded operators for types `annotated_ref<T, ...>` and
+`annotated_ref<T2, ...>`. Let `operatorOP` denote the operator used.
+The overloaded operator `operatorOP` utilizes `operatorOP(T, T2)` and
+is available only if `operatorOP(T, T2)` is well formed. The value and result
+is the same as the result of `operatorOP(T, T2)` applied to the objects of
+type `T` and `T2`.
```

Comment:
Roland suggested to use the hidden friend function which seems more elegant. Adopted the implementation and updated the spec.

I think we just need to mention the `load` right? Since there should be no store here since we do return-by-value in the binary operators.